dlmalloc/src/malloc.c

```diff
@@ -5227,23 +5227,16 @@ static void try_init_allocator(void) {
 
   char *base = &__heap_base;
   // Try to use the linker pseudo-symbol `__heap_end` for the initial size of
-  // the heap, but if that's not defined due to LLVM being too old perhaps then
-  // round up `base` to the nearest `PAGESIZE`. The initial size of linear
-  // memory will be at least the heap base to this page boundary, and it's then
-  // assumed that the initial linear memory image was truncated at that point.
-  // While this reflects the default behavior of `wasm-ld` it is also possible
-  // for users to craft larger linear memories by passing options to extend
-  // beyond this threshold. In this situation the memory will not be used for
-  // dlmalloc.
-  //
-  // Note that `sbrk(0)`, or in dlmalloc-ese `CALL_MORECORE(0)`, is specifically
-  // not used here. That captures the current size of the heap but is only
-  // correct if the we're the first to try to grow the heap. If the heap has
-  // grown elsewhere, such as a different allocator in place, then this would
-  // incorrectly claim such memroy as our own.
+  // the heap.
   char *end = &__heap_end;
-  if (end == NULL)
-    end = (char*) page_align((size_t) base);
+  if (end == NULL) {
+    // This can happen when 1. you are using an old wasm-ld which doesn't
```

Comment:
This (1) case is not true anymore I think.  Older versions of llvm will simply cause link failure, not result in a zero-addressed version of `__heap_end`.

Reply:
I don't think we need to worry that much about it, since it will only effect people that are both (a) using the old linker and (b) are weakly referencing `__heap_end` elsewhere in their program.   I think this is highly unlikely isn't it?   

Maybe at least leave a TODO to completely remove this check once we drop support for older wasm-ld?

Reply:
Just to be clear, is https://reviews.llvm.org/D144747 actually required in order to guarantee that this symbol will be non-NULL?  Or is it simply enough to have 15.0.7?     IIUC if you have 15.0.7 then this can never be null, so this check is only in support of < 15.0.7?